test/e2e/i18n-default-locale-redirect/i18n-default-locale-redirect.test.ts

```diff
@@ -0,0 +1,44 @@
+import { join } from 'path'
+import { createNextDescribe, FileRef } from 'e2e-utils'
+
+createNextDescribe(
+  'i18-default-locale-redirect',
+  {
+    files: {
+      pages: new FileRef(join(__dirname, './app/pages')),
+      'next.config.js': new FileRef(join(__dirname, './app/next.config.js')),
+    },
+  },
+  ({ next }) => {
+    it('should not request a path prefixed with default locale', async () => {
+      const browser = await next.browser('/')
+      let requestedDefaultLocalePath = false
+      browser.on('request', (request: any) => {
+        if (new URL(request.url(), 'http://n').pathname === '/en/to-new') {
+          requestedDefaultLocalePath = true
+        }
+      })
+
+      await browser.elementByCss('#to-new').click().waitForElementByCss('#new')
+      expect(await browser.elementByCss('#new').text()).toBe('New')
+      expect(requestedDefaultLocalePath).toBe(false)
```

Comment:
Confirmed that this test fails with the current `canary`:

```
 FAIL  test/e2e/i18n-default-locale-redirect/i18n-default-locale-redirect.test.ts (17.271 s)
  i18-default-locale-redirect
    ✕ should not request a path prefixed with default locale (899 ms)
    ✓ should request a path prefixed with non-default locale (150 ms)

  ● i18-default-locale-redirect › should not request a path prefixed with default locale

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      22 |       await browser.elementByCss('#to-new').click().waitForElementByCss('#new')
      23 |       expect(await browser.elementByCss('#new').text()).toBe('New')
    > 24 |       expect(requestedDefaultLocalePath).toBe(false)
         |                                          ^
      25 |     })
      26 |
      27 |     it('should request a path prefixed with non-default locale', async () => {

      at Object.<anonymous> (e2e/i18n-default-locale-redirect/i18n-default-locale-redirect.test.ts:24:42)

Reply:
## Failure Demo (current `canary`)

Clicking on `<Link href="/to-new" id="to-new">` hard-navigates to `/en/to-new` instead of `/to-new` (before redirecting to `/new`):

https://user-images.githubusercontent.com/992008/225452607-9b9a055a-0854-4644-b2ba-1d8681862542.mp4

Reply:
## Success Demo (after my changes)

It now hard-navigates to `/to-new` instead, omitting `defaultLocale`.

https://user-images.githubusercontent.com/992008/225453322-575b736b-0767-477a-89e1-b950a2cdc0d1.mp4